src/main/java/ch/njol/skript/lang/BukkitOrigin.java

```diff
@@ -0,0 +1,74 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.lang;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.Contract;
+import org.skriptlang.skript.registry.SyntaxOrigin;
+
+public final class BukkitOrigin implements SyntaxOrigin {
+	
+	@Contract("_ -> new")
+	public static SyntaxOrigin of(Plugin plugin) {
+		return new BukkitOrigin(plugin);
```

Comment:
instead of having a Plugin constructor, do you think it'd be better to just do something like:
```suggestion
		return new BukkitOrigin(plugin.getClass().getCanonicalName());
```

Reply:
I disagree since in the end we would like a Class<?> for the origin, not possible because of breaking changes but it should be the end goal.

Reply:
For it should be fine though, so I will commit it.